frame/contracts/src/schedule.rs

```diff
@@ -140,101 +132,15 @@ impl Limits {
-///    individual values to derive (by subtraction) the weight of all other instructions
-///    that use them as supporting instructions. Supporting means mainly pushing arguments
-///    and dropping return values in order to maintain a valid module.
+/// Gas metering of Wasm executed instructions is being done on the engine side.
+/// This struct holds a reference value used to gas units scaling between host and engine.
 #[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
 #[derive(Clone, Encode, Decode, PartialEq, Eq, ScheduleDebug, TypeInfo)]
 #[scale_info(skip_type_params(T))]
 pub struct InstructionWeights<T: Config> {
-	/// Version of the instruction weights.
-	///
-	/// # Note
-	///
-	/// Should be incremented whenever any instruction weight is changed. The
-	/// reason is that changes to instruction weights require a re-instrumentation
-	/// in order to apply the changes to an already deployed code. The re-instrumentation
-	/// is triggered by comparing the version of the current schedule with the version the code was
-	/// instrumented with. Changes usually happen when pallet_contracts is re-benchmarked.
-	///
-	/// Changes to other parts of the schedule should not increment the version in
-	/// order to avoid unnecessary re-instrumentations.
-	pub version: u32,
-	/// Weight to be used for instructions which don't have benchmarks assigned.
-	///
-	/// This weight is used whenever a code is uploaded with [`Determinism::Relaxed`]
-	/// and an instruction (usually a float instruction) is encountered. This weight is **not**
-	/// used if a contract is uploaded with [`Determinism::Enforced`]. If this field is set to
-	/// `0` (the default) only deterministic codes are allowed to be uploaded.
-	pub fallback: u32,
-	pub i64const: u32,
-	pub i64load: u32,
-	pub i64store: u32,
-	pub select: u32,
-	pub r#if: u32,
-	pub br: u32,
-	pub br_if: u32,
-	pub br_table: u32,
-	pub br_table_per_entry: u32,
-	pub call: u32,
-	pub call_indirect: u32,
-	pub call_per_local: u32,
-	pub local_get: u32,
-	pub local_set: u32,
-	pub local_tee: u32,
-	pub global_get: u32,
-	pub global_set: u32,
-	pub memory_current: u32,
-	pub memory_grow: u32,
-	pub i64clz: u32,
-	pub i64ctz: u32,
-	pub i64popcnt: u32,
-	pub i64eqz: u32,
-	pub i64extendsi32: u32,
-	pub i64extendui32: u32,
-	pub i32wrapi64: u32,
-	pub i64eq: u32,
-	pub i64ne: u32,
-	pub i64lts: u32,
-	pub i64ltu: u32,
-	pub i64gts: u32,
-	pub i64gtu: u32,
-	pub i64les: u32,
-	pub i64leu: u32,
-	pub i64ges: u32,
-	pub i64geu: u32,
-	pub i64add: u32,
-	pub i64sub: u32,
-	pub i64mul: u32,
-	pub i64divs: u32,
-	pub i64divu: u32,
-	pub i64rems: u32,
-	pub i64remu: u32,
-	pub i64and: u32,
-	pub i64or: u32,
-	pub i64xor: u32,
-	pub i64shl: u32,
-	pub i64shrs: u32,
-	pub i64shru: u32,
-	pub i64rotl: u32,
-	pub i64rotr: u32,
+	/// Base instruction `ref_time` Weight.
+	/// Should match to wasmi's `1` fuel (see <https://github.com/paritytech/wasmi/issues/701>).
+	pub base: u32,
```

Comment:
isn't unit a better name, since if I understand correctly this represent the weight of 1 unit of wasm fuel
or weight_to_wasmi_fuel ?

Reply:
I just made it compliant with the wasmi terminology, where it's called *base cost* of a simple instruction